### ISSUE TITLE
Tracker: Running list

### ISSUE THREAD

**Issue by rmusser01** [rmusser01/tldw_server#599]
Things to do:

Config
- [x] Centralize all DBs into a proper folder in the config.

Local LLM Management/Inference
- [ ] Review, Validation + Tests

Media
- [ ] Fan-out processing for audio/video

Chat
- [x] Validate _all_ API endpoints
- [x] Write tests for image uploads/processing
- [ ] Update HF API to match current
- [x] Support for 'extra' options (logits, samplers, etc)
- [ ] Validate/Test MediaWiki Endpoints/Library
- [ ] Setup 'Workflows' endpoint + storage/management of workflows

Prompts
- [x] Setup Prompts Endpoint
- [x] Setup Prompts DB
- [ ] Setup Syncing for prompts DB
- [x] Setup Structured Prompt Creation Endpoint

Research Endpoint
- [ ] Setup 'Research' Endpoint
- [x] Setup Arxiv
- [ ] Setup PubMed
- [x] Setup Semantic Scholar

Tools
- [ ] Setup 'Tools' Endpoint
- [ ] Setup Anki Endpoint

Sync
- [x] Setup client sync library
- [ ] Setup Server Sync library
- [ ] Setup `/sync` endpoint
- [ ] Validate + test extensively `/sync` endpoint (tests + models + extensive usage/identification of edge cases)
- [ ] Setup `when a client sees an update to a Media item, it must: Delete all local UnvectorizedMediaChunks for that media_id. Then apply any create operations for UnvectorizedMediaChunks (for that media_id) from the current sync batch.`
- [ ] Create issue for looking into synching character card images

Audio/Speech
- [x] Setup `/v1/audio/* ` endpoint ala openai, points to existing process-audio endpoint https://platform.openai.com/docs/api-reference/audio
- [ ] /v1/audio/speech
- [x] /v1/audio/transcriptions - front for existing process-audio endpoint
- [ ] /v1/audio/translations
- [ ] Write new tests for all this stuff 
- [ ] Go through and fix placeholder stuff

Embeddings + Vector Store Management
- [x] Copy OpenAI API https://platform.openai.com/docs/api-reference/embeddings
- [ ] Copy OpenAI API: https://platform.openai.com/docs/api-reference/vector-stores

RAG Endpoint - /retrieval
- [x] custom endpoint for RAG queries, uses RAG library to interact with DB + Embeddings to perform search + Reranking + Retrieval
- [ ] Option to create embeddings if they don't already exist for searched media items/groups/keywords (Warning about time)
- [ ] [Setup Proper Endpoint for 'Workspaces/Vector Collections/IDK'](https://platform.openai.com/docs/api-reference/vector-stores/create)
- [ ] f

Evals
- [ ] Copy OpenAI API: https://platform.openai.com/docs/api-reference/evals

Responses API Endpoint
- [ ] Copy OpenAI Responses API: https://platform.openai.com/docs/api-reference/responses
- [ ] Extend to match needs

Graders
- [ ] Copy OpenAI https://platform.openai.com/docs/api-reference/graders
- [ ] F

Re-ranker endpoint
- [ ] setup an endpoint to perform (re-)ranking of submitted items
Search
- [ ] Setup `/search` endpoint
- [ ] Validate `search` endpoint works (tests + models)

Multi-User Setup
- [ ] Setup User DB Schema
- [ ] Setup user registration
- [ ] Setup user registration endpoint
- [ ] Setup token validation/checking
- [ ] Validate token validation/checking
